src/pages/md-asset-depreciation/index.js

```diff
@@ -0,0 +1,188 @@
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useResourceQuery } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Fixed } from 'src/components/Shared/Layouts/Fixed'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useWindow } from 'src/windows'
+import { ControlContext } from 'src/providers/ControlContext'
+import RPBGridToolbar from 'src/components/Shared/RPBGridToolbar'
+import { FixedAssetsRepository } from 'src/repositories/FixedAssetsRepository'
+import AssetsForm from './form/AssetsForm'
+
+const AssetsDescription = () => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+  const [params, setParams] = useState('')
+
+  const { stack } = useWindow()
+
+  async function fetchGridData(options = {}) {
+    const { _startAt = 0, _pageSize = 50, params = [] } = options
+
+    const response = await getRequest({
+      extension: FixedAssetsRepository.AssetsDescription.qry,
+      parameters: `_startAt=${_startAt}&_pageSize=${_pageSize}&_filter=&_sortField=&_params=${params}`
+    })
+
+    return { ...response, _startAt: _startAt }
+  }
+
+  const {
+    query: { data },
+    filterBy,
+    refetch,
+    clearFilter,
+    labels: _labels,
+    access,
+    paginationParameters,
+    invalidate
+  } = useResourceQuery({
+    queryFn: fetchGridData,
+    endpointId: FixedAssetsRepository.AssetsDescription.snapshot,
+    datasetId: ResourceIds.Depreciation,
+    filter: {
+      filterFn: fetchWithFilter
+    }
+  })
+  async function fetchWithFilter({ filters, pagination = {} }) {
+    const { _startAt = 0, _size = 50 } = pagination
+    if (filters.qry) {
+      const response = await getRequest({
+        extension: FixedAssetsRepository.AssetsDescription.snapshot,
+        parameters: `_filter=${filters.qry}&_startAt=${_startAt}&_size=${_size}`
+      })
+
+      return { ...response, _startAt: _startAt }
+    } else return fetchGridData({ _startAt: pagination._startAt || 0, params: filters?.params })
+  }
+
+  const columns = [
+    {
+      field: 'dtName',
+      headerName: _labels.dtName,
+      flex: 1
+    },
+    {
+      field: 'reference',
+      headerName: _labels.reference,
+      flex: 1
+    },
+    {
+      field: 'date',
+      headerName: _labels.date,
+      flex: 1,
+      type: 'date'
+    },
+    {
+      field: 'plantName',
+      headerName: _labels.plantName,
+      flex: 1
+    },
+    {
+      field: 'statusName',
+      headerName: _labels.status,
+      flex: 1
+    },
+    {
+      field: 'wipName',
+      headerName: _labels.wip,
+      flex: 1
+    },
+    {
+      field: 'notes',
+      headerName: _labels.notes,
+      flex: 1
+    }
+  ]
+
+  function openForm(obj) {
+    stack({
+      Component: AssetsForm,
+      props: {
+        labels: _labels,
+        obj,
```

Comment:
why u are passing the whole object to retrieve only recordId?, pass recordId directly